diff --git a/src/Test/Cardano/Ledger/Generic/Functions.hs b/src/Test/Cardano/Ledger/Generic/Functions.hs
index 9fd6c15936..e71bc9791c 100644
--- a/src/Test/Cardano/Ledger/Generic/Functions.hs
+++ b/src/Test/Cardano/Ledger/Generic/Functions.hs
@@ -51,7 +51,7 @@ import Data.Maybe (maybeToList)
 import Data.Maybe.Strict (StrictMaybe (..))
 import Data.Set (Set)
 import qualified Data.Set as Set
-import Lens.Micro
+import Lens.Micro --
 import Numeric.Natural
 import Test.Cardano.Ledger.Alonzo.Arbitrary (alwaysFailsLang, alwaysSucceedsLang)
 import Test.Cardano.Ledger.Generic.ModelState (MUtxo, Model, ModelNewEpochState (..))
