diff --git a/cabal.project b/cabal.project
index 3d18d880d8..75065508c2 100644
--- a/cabal.project
+++ b/cabal.project
@@ -89,6 +89,20 @@ allow-newer:
 -- Do NOT add more source-repository-package stanzas here unless they are strictly
 -- temporary! Please read the section in CONTRIBUTING about updating dependencies.
 
+source-repository-package
+  type: git
+  location: https://github.com/IntersectMBO/cardano-ledger.git
+  tag: 94e9618c91a16ec08db477632a158b630722089b
+  subdir:
+    libs/cardano-ledger-test
+  --sha256: sha256-BgeRHvwhrtDnce4nGRd/S/PME8uD3Pxrb5rXe7sMlqw=
+
+source-repository-package
+  type: git
+  location: https://github.com/input-output-hk/constrained-generators.git
+  --sha256: sha256-h0bJTwdfJG84XrERJJoBBi5ZzRHgeVoZWS2yW/z7LiU=
+  tag: 966f65f3c28cb22fc45220dbd1408baea93b53d8
+
 if impl(ghc >= 9.12)
   -- GHC 9.12 support - master branch
   source-repository-package
diff --git a/cardano-node/cardano-node.cabal b/cardano-node/cardano-node.cabal
index da3d9cc7b0..4b8f6dd31a 100644
--- a/cardano-node/cardano-node.cabal
+++ b/cardano-node/cardano-node.cabal
@@ -153,6 +153,7 @@ library
                       , cardano-ledger-core
                       , cardano-ledger-dijkstra
                       , cardano-ledger-shelley
+                      , cardano-ledger-test == 9.9.9.9
                       , cardano-prelude
                       , cardano-protocol-tpraos >= 1.4
                       , cardano-slotting >= 0.2
diff --git a/nix/haskell.nix b/nix/haskell.nix
index cf82e63384..c49110ee39 100644
--- a/nix/haskell.nix
+++ b/nix/haskell.nix
@@ -146,6 +146,14 @@ let
       #
       modules =
         [
+          ({ lib, pkgs, ... }: {
+            packages.cardano-ledger-test = {
+              # src = CARDANO_LEDGER_SOURCE;
+              # subdir = "libs/cardano-ledger-test";
+              doCheck = false;
+              #doBenchmark = false;
+            };
+          })
           ({ lib, pkgs, ... }: {
             packages.cardano-ledger-core.patches = [
               ./cardano-ledger-core--Arbitrary-PoolMetadata.diff
